### DISCUSSION TITLE
Please download LDSR to activate more upscale features...

### DISCUSSION THREAD

**Discussion by sedetweiler** [hlky/stable-diffusion-webui#450] [Q&A]
The instructions say... "Please download LDSR to activate more upscale features, instructions are available at the [Github](https://github.com/hlky/stable-diffusion-webui)"  But nowhere on the site is there any sort of a hint on what to do next.

**Comment by AmoArt** [comment]
+1 to this, I'm going to assume that author simply forgot to update the instructions when updating the code.
Additional, I would also love to know were to get the 'GoLatent' model since that is not included either too.

**Reply by devilismyfriend** [reply]
> +1 to this, I'm going to assume that author simply forgot to update the instructions when updating the code. Additional, I would also love to know were to get the 'GoLatent' model since that is not included either too.

GoLatent is just using GoBig before LDSR, it's not a separate model.

**Comment by JoshuaKimsey** [comment]
Yeah, I was just fixing to open a discussion for this. @hlky and @devilismyfriend, we're gonna need some instructions on how to activate the LDSR part of the new upscalers addition, like we have for GFPGAN and realESRGAN. I know in the PR it mentioned needing to clone a repo and download a model, but not sure if that held through to the final merge.

Also wanna know how this works so I can update my Linux script to handle this automatically for a user. :)

**Reply by darkstorm2150** [reply]
Need github Instruction to install Latent Diffusion Super Resolution :D

**Reply by mchaker** [reply]
It looks like this is the way to get LDSR:

https://github.com/devilismyfriend/latent-diffusion#get-the-models-1

**Comment by devilismyfriend** [comment]
> The instructions say... "Please download LDSR to activate more upscale features, instructions are available at the [Github](https://github.com/hlky/stable-diffusion-webui)" But nowhere on the site is there any sort of a hint on what to do next.

Yes lol it's a placeholder, sorry, as you can tell this is in active development still.

How to install LDSR:

git clone https://github.com/devilismyfriend/latent-diffusion to your src folder, make sure the name of the folder is "latent-diffusion" and not "latent-diffusion-main".

Go into the latent-diffusion folder and you can use the download_model.bat to download and place the models or you can do it yourself by placing the model.ckpt and project.yaml files under /src/latent-diffusion/experiments/pretrained_models/

**Reply by hlky** [reply]
can just add this [this](https://github.com/devilismyfriend/latent-diffusion) to environment.yaml 
will do it soon, testing the new streamlit version

**Reply by devilismyfriend** [reply]
cool thanks, I changed the readme and wrote down to git clone the repo, but once you add it to the environment we'll just remove that part.

**Reply by churchofthought** [reply]
thanks guys, yall deserve some 👏

**Reply by sedetweiler** [reply]
> thanks guys, yall deserve some 👏

I actually gave some cash. Thanks again, guys!

**Comment by AmoArt** [comment]
the devilismyfriend instructions work, however when running the upscale process 'Latent Diffusion SR' it ignores/overrides my sampling method and steps and sets them up as DDIM 100 steps each run.

**Reply by TristanMorelle** [reply]
@devilismyfriend 
  with single and multiple images i get this error
  nothing was produced
  
  
  
```
  Processing images...
Unloaded GFPGAN
Loaded RealESGAN
GoBig upscaling will process a total of 9 images tiled as 3x3 in a total of 9 batches.
[MemMon] Recording max memory usage...

Iteration: 1/1
100%|██████████████████████████████████████████████████████████████████████████████████| 16/16 [00:01<00:00,  8.08it/s]
Traceback (most recent call last):
  File "C:\Users\Tristan\.conda\envs\ldo\lib\site-packages\gradio\routes.py", line 247, in run_predict
    output = await app.blocks.process_api(
  File "..\.conda\envs\ldo\lib\site-packages\gradio\blocks.py", line 641, in process_api
    predictions, duration = await self.call_function(fn_index, processed_input)
  File "..\.conda\envs\ldo\lib\site-packages\gradio\blocks.py", line 556, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "..\.conda\envs\ldo\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File ..\.conda\envs\ldo\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File ..\.conda\envs\ldo\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "scripts/webui.py", line 1731, in imgproc
    image = processGoBig(image)
  File "scripts/webui.py", line 1665, in processGoBig
    work_results.append(output_images[0])
IndexError: list index out of range
[MemMon] Stopped recording.
```

**Comment by bilalqureshi1992** [comment]
even after the LDSR part it's giving me the following error, (any solution??):
RuntimeError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 4.00 GiB total capacity; 3.42 GiB already allocated; 0 bytes free; 3.48 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

**Reply by mchaker** [reply]
Your graphics card is not powerful enough (not enough VRAM).

Have you tried the Doggettx memory optimization? : https://gist.github.com/mchaker/e0cf180db6849fa03696293b3fc8f485

**Reply by StantheBrain** [reply]
-----RUNNING ON 4GB-----  https://rentry.org/GUItard
1- Edit /scripts/relauncher.py
2- Edit line 14
(# Run Stable Diffusion in Optimized Mode - Only requires 4Gb of VRAM, but is significantly slower) 
Line 14 change : False to True 
3- Save 

(this tells it to optimize Vram by generating incrementally, it should be noticed that this sacrifices generation speed)

**Reply by bilalqureshi1992** [reply]
thanks man that worked like a charm :)

**Comment by devilismyfriend** [comment]
LDSR requires at least 8GB of ram.

**Reply by bilalqureshi1992** [reply]
ok if I don't want to use it, how can I comment it out?

**Comment by devilismyfriend** [comment]
Just don't use it? It's hardly a default option
